### PR TITLE
Replace `storage.fetch_ordered_annotation` with an `AnnotationReadService` method

### PR DESCRIPTION
**Pull request by jon-betts**
For:

 * https://github.com/hypothesis/h/issues/7932
 
Based on:

 * https://github.com/hypothesis/h/pull/7947

## Testing notes

 * Start H, the client, Via and ViaHTML
 * `make devdata`
 * Login to H: http://0.0.0.0:5000/login (devdata_user / pass)
 * Visit: http://localhost:9083/https://example.com/
 * Make an annotation with some text which is easy to spot

Checking search:

 * Visit: http://0.0.0.0:5000/users/devdata_user
 * Expand the annotation on example.com
 * You should see your annotation content
 
Checking annotation read JSON in the client:
 
 * Refresh the page: http://localhost:9083/https://example.com/
 * You should see the annotations show up
 * In the background we load: http://localhost:5000/api/search?limit=50&sort=created&order=asc&_separate_replies=false&group=__world__&uri=https%3A%2F%2Fexample.com%2F

Checking Atom and RSS feeds

 * Visit: http://0.0.0.0:5000/stream?q=
 * Visit: http://0.0.0.0:5000/stream.atom
 * Visit: http://0.0.0.0:5000/stream.rss
 * Your annotation should show up in each case

### CODE REVIEW

File: h/services/annotation_read.py [hunk 1]
```diff
@@ -23,6 +24,42 @@ def get_annotation_by_id(self, id_: str) -> Optional[Annotation]:
...
+        """
+
+        if not ids:
+            return []
+
+        annotations = self._db.execute(
+            self._annotation_search_query(ids=ids, eager_load=eager_load)
+        ).scalars()
+
+        return sorted(annotations, key=lambda annotation: ids.index(annotation.id))
```
Comment (marcospri) [inline h/services/annotation_read.py:45]: ~Is ordereding by ID not something postgres could do for us?~

Ah I get it now, sorted by the input ids :+1:

**Review by marcospri — APPROVED** [review]
LGTM, this looks like the  right direction to set up thigns for bigger chages.

File: h/activity/query.py [hunk 2]
```diff
@@ -155,16 +155,11 @@ def aggregations_for(query):
 
 
 @newrelic.agent.function_trace()
```
[the remark this answers is not in the document]
Comment (jon-betts) [reply h/activity/query.py:157]: I'd like to go over this all in a second pass and find places where we query for annotations outside of the services.

File: h/services/annotation_json.py [hunk 3]
```diff
@@ -133,34 +140,25 @@ def present_all_for_user(self, annotation_ids, user: User):
         self._flag_service.all_flagged(user, annotation_ids)
         self._flag_service.flag_counts(annotation_ids)
 
-        annotations = storage.fetch_ordered_annotations(
-            self._session,
-            annotation_ids,
-            query_processor=self._eager_load_related_items,
+        annotations = self._annotation_read_service.get_annotations_by_id(
```
[the remark this answers is not in the document]
Comment (jon-betts) [reply h/services/annotation_json.py:143]: Yeah, there's some divergence of opinion here. Sean was suggesting the opposite and unifying under a single method.

**Pull request merged**